packages/firestore/src/remote/datastore.ts

```diff
@@ -149,6 +150,33 @@ export async function invokeBatchGetDocumentsRpc(
   return result;
 }
 
+export async function invokeRunQueryRpc(
+  datastore: Datastore,
+  query: Query
+): Promise<Document[]> {
+  const datastoreImpl = debugCast(datastore, DatastoreImpl);
+  const { structuredQuery, parent } = datastoreImpl.serializer.toQueryTarget(
+    query.toTarget()
+  );
+  const params = {
+    database: datastoreImpl.serializer.encodedDatabaseId,
+    parent,
+    structuredQuery
+  };
+
+  const response = await datastoreImpl.invokeStreamingRPC<
+    api.RunQueryRequest,
+    api.RunQueryResponse
+  >('RunQuery', params);
+
+  return (
+    response
+      // Filter RunQueryResponses that only contain readTimes.
```

Comment:
Nit: from reading the PR description, it appears that only the last element may not contain a `document`, whereas the comment makes it seem as if there might be several such elements throughout the container.

Reply:
The PR description reflects the behavior that I have observed. The protocol state though that it is also valid to not provide a document mid-stream as a way to report status.